File: go/batch_job_response.go
// this file is @generated
package meteroid

import "time"

type BatchJobResponse struct {
	// RFC 3339 timestamp.
	CompletedAt *time.Time `json:"completed_at,omitempty"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	CreatedBy string `json:"created_by"`

	FailedItems int32 `json:"failed_items"`

	Id BatchJobId `json:"id"`

	InputFileName *string `json:"input_file_name,omitempty"`

	JobType BatchJobType `json:"job_type"`

	ProcessedItems int32 `json:"processed_items"`

	Status BatchJobStatus `json:"status"`

	TotalItems *int32 `json:"total_items,omitempty"`
}

File: python/meteroid/models/batch_job_response.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .batch_job_id import BatchJobId
from .batch_job_status import BatchJobStatus
from .batch_job_type import BatchJobType


@dataclasses.dataclass
class BatchJobResponse(BaseModel):
    created_at: datetime

    created_by: str

    failed_items: int

    id: BatchJobId

    job_type: BatchJobType

    processed_items: int

    status: BatchJobStatus

    completed_at: t.Optional[datetime] = None

    input_file_name: t.Optional[str] = None

    total_items: t.Optional[int] = None
